raiden-dapp/src/utils/logstore.ts

```diff
@@ -98,14 +99,19 @@ export async function setupLogStore(additionalLoggers: string[] = ['matrix']): P
 
   for (const log of [logging, ...additionalLoggers.map(logging.getLogger)]) {
     const origFactory = log.methodFactory;
-    log.methodFactory = function raidenMethodFactory(methodName, level, loggerName) {
+
+    function raidenMethodFactory(
+      methodName: string,
+      level: LogLevelNumbers,
+      loggerName: string | symbol,
+    ): LoggingMethod {
       const rawMethod = origFactory(methodName, level, loggerName);
       const name = loggerName.toString();
       setDbForLogger(name);
 
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       return (...message: any[]): void => {
-        rawMethod(...message);
+        rawMethod.apply({}, [...message]);
```

Comment:
Could this have any unintended consequences?

Reply:
To be honest, I'm not 100% sure. From reading the libraries code around [this location](https://github.com/pimterry/loglevel/blob/342f103eed48f29db854533d2cbfe254690f9113/lib/loglevel.js#L84) it looks like this should be fine. Practical tests seem to have proven that it works. Combining these two I'm optimistic its fine. But maybe you see something?

Reply:
I also don't found any further details in the issues.

Reply:
I just see that in the CLI we put in our own this. So maybe we just do that here as well.

Reply:
Alright, I have a better solution. 👍🏾

Reply:
Together with link I think this is fine.